extensions/amp-youtube/0.1/amp-youtube.js

```diff
@@ -71,6 +91,57 @@ class AmpYoutube extends AMP.BaseElement {
     // the playback.
     return false;
   }
+
+  /** @private */
+  buildImagePlaceholder_() {
+    const imgPlaceholder = new Image();
+    const videoid = this.videoid_;
+
+    // Because sddefault.jpg isn't available for all videos, we try to load
+    // it and fallback to hqdefault.jpg.
+    loadPromise(imgPlaceholder).then(function waitTillReady() {
+      // Wait until the image has completed.
+      if (!imgPlaceholder.complete) {
+        return timer.promise(10).then(waitTillReady);
```

Comment:
To be honest, I'm not a huge fun of recursive call here. We have to options: 
 1. Have another promise chain that will poll until `complete` is there, but with a time constraint.
 2. Just sleep once for 10ms and then risk it not being ready.

If you want to poll, this can be arranged like this:
```
loadPromise(imagePlaceholder).then(() => {
  // Poll until complete is set.
  if (!imgPlaceholder.complete) {
    return new Promise(resolve => {
      const maxTime = timer.now() + 100;
      const interval = setInterval(() => {
        if (imgPlaceholder.complete || timer.now() > maxTime) {
          clearInterval(interval);
          // Here I'm interpreting timeout as "resolve". But you can always do "reject".
          resolve();
        }
      }, 10);
    });
  }
}).then(() => {
  // process
})...
```

But ideally I'd like us to understand why `complete` is not available yet. Are there some other events we should be listening to? We can certainly work things around, but we should try to get a better idea why this is happening.